src/System.Net.Http/tests/StressTests/HttpStress/ClientOperations.cs

```diff
@@ -23,60 +23,61 @@ public sealed class RequestContext
         {
             _random = random;
             _client = httpClient;
-            _cancellationProbability = cancellationProbability;
-            _http2Probability = http2Probability;
+            _cancellationProbability = config.CancellationProbability;
+            _globalToken = globalToken;
 
             TaskNum = taskNum;
+            HttpVersion = config.HttpVersion;
             IsCancellationRequested = false;
-            MaxRequestParameters = maxRequestParameters;
-            MaxRequestUriSize = maxRequestUriSize;
-            ContentSource = contentSource;
+            MaxRequestParameters = config.MaxParameters;
+            MaxRequestUriSize = config.MaxRequestUriSize;
+            MaxContentLength = config.MaxContentLength;
         }
 
         public int TaskNum { get; }
+        public Version HttpVersion { get; }
         public bool IsCancellationRequested { get; set; }
-        public string ContentSource { get; }
         public int MaxRequestParameters { get; }
         public int MaxRequestUriSize { get; }
-        public int MaxContentLength => ContentSource.Length;
+        public int MaxContentLength { get; }
         public Uri BaseAddress => _client.BaseAddress;
 
         // HttpClient.SendAsync() wrapper that wires randomized cancellation
         public async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, HttpCompletionOption httpCompletion = HttpCompletionOption.ResponseContentRead, CancellationToken? token = null)
         {
+            request.Version = HttpVersion;
+
             if (token != null)
             {
                 // user-supplied cancellation token overrides random cancellation
-                return await _client.SendAsync(request, httpCompletion, token.Value);
+                using var cts = CancellationTokenSource.CreateLinkedTokenSource(_globalToken, token.Value);
+                return await _client.SendAsync(request, httpCompletion, cts.Token);
             }
             else if (GetRandomBoolean(_cancellationProbability))
             {
                 // trigger a random cancellation
-                using (var cts = new CancellationTokenSource())
-                {
-                    int delayMs = _random.Next(0, 2);
-                    Task<HttpResponseMessage> task = _client.SendAsync(request, httpCompletion, cts.Token);
-                    if (delayMs > 0)
-                        await Task.Delay(delayMs);
-
-                    cts.Cancel();
-                    IsCancellationRequested = true;
-                    return await task;
-                }
+                using var cts = CancellationTokenSource.CreateLinkedTokenSource(_globalToken);
+
+                int delayMs = _random.Next(0, 2);
+                Task<HttpResponseMessage> task = _client.SendAsync(request, httpCompletion, cts.Token);
+                if (delayMs > 0)
+                    await Task.Delay(delayMs);
```

Comment:
I know your PR isn't changing this and we've talked about it before, but ...

This is either going to not wait at all, or wait for 1ms, which means on Windows closer to 15ms, which is very likely going to be after the request has already completed.  Rather than using Task.Delay, maybe we should just do a random small amount of spinning.

Reply:
@stephentoub I took the time to measure latencies for the stress suite, here's one result:
```
n=95511, p50=1.10, p99=59.46, p999=97.80, max=1011.88
```
While the median request sits squarely within the 1ms mark, 1% of requests is ~60ms or above. I'd argue we'd need to delay at least some of the cancellations, if only to trigger potentially interesting behaviour in these outliers.

I'd say we could randomly spinwait for 50% of requests, then delay the rest of them.

Reply:
That's fine, thanks.